lib/events/index.js

```diff
@@ -2073,15 +2083,25 @@ export default function (self) {
           ? cellsLength - colPosReal - 1
           : colPosReal;
         const columnIndex = startColumnIndex + colPosition;
-        var column = schema[columnIndex];
-
-        if (!column) {
-          console.warn('Paste data exceeded grid bounds. Skipping.');
-          continue;
+        if (!self.getSchema()[columnIndex]) {
+          if (self.attributes.allowGridSizeChangeOnPaste) {
+            const lastColSchema = self.getSchema()[
+              self.orders.columns[self.getSchema().length - 1]
+            ];
+            const newColSchema = {
+              ...lastColSchema,
+              name: `col${self.schema.length + 1}:${Date.now()}`,
```

Comment:
Should we not be using `self.getSchema()` here? Or rather:

```js
const schema = self.getSchema();

if (!schema[columnIndex]) {
  if (self.attributes.allowGridSizeChangeOnPaste) {
    const lastColSchema = schema[self.orders.columns[schema.length - 1]];
  }
  const newColSchema = {
    ...lastColSchema,
    name: `col${schema.length + 1}${Date.now()}`,
    title: ' ',
  };
  self.addColumn(newColSchema)
}
```

Also, two (three?) other points:

1) can we safely assume that the last column's schema is similar to the newly pasted column? What is you're pasting text and the previous existing column schema defines it as `type: number` ?
2) What is the purpose of the name including the current date? Do we even need to name the new column?
3) Why the empty title?